File: 4313-password-strength/password-strength.java
class Solution {
    public int passwordStrength(String password) {
        HashSet<Character> lower=new HashSet<>();
        HashSet<Character> upper=new HashSet<>();
        HashSet<Character> digit=new HashSet<>();
        HashSet<Character> special=new HashSet<>();
        for(char ch:password.toCharArray()){
            if(Character.isLowerCase(ch)){
                lower.add(ch);
            }
            else if(Character.isUpperCase(ch)){
                upper.add(ch);
            }
            else if(Character.isDigit(ch)){
                digit.add(ch);
            }
            else{
                special.add(ch);
            }
        }
        int strength=(1*lower.size())+(2*upper.size())+(3*digit.size())+(5*special.size());
        return strength;
    }
}

File: 4313-password-strength/password-strength.cpp
class Solution {
public:
    int passwordStrength(string password) {
        unordered_set<char> lower;
        unordered_set<char> upper;
        unordered_set<char> digit;
        unordered_set<char> special;
        for(char ch:password){
            if(islower(ch)){
                lower.insert(ch);
            }
            else if(isupper(ch)){
                upper.insert(ch);
            }
            else if(isdigit(ch)){
                digit.insert(ch);
            }
            else{
                special.insert(ch);
            }
        }
        int strength=(1*lower.size())+(2*upper.size())+(3*digit.size())+(5*special.size());
        return strength;
    }
};